### PR TITLE
Update config and run Detekt during the build

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This changeset updates `config` and includes Detekt code analysis into the build.

Also, I've noticed the presence of `pull` script directly in root directories of `base`, `base-types` and `time` repos:

```kotlin
cd ./config || exit
echo "Switching the pointer of the 'config' submodule to its 'master'."
git checkout master
git pull

cd ..
./config/pull
```

This script eliminates the need to run `./config/pull` twice when one knows there are changes in `pull` script itself.

Can we have one in other repos? Or even include it into `template` repo?

The version is bumped to `2.0.0-SNAPSHOT.118`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,11 @@ all modules and does not describe the project structure per-subproject.
     <artifactId>checkstyle</artifactId>
     <version>10.3.4</version>
   </dependency>
+  <dependency>
+    <groupId>io.gitlab.arturbosch.detekt</groupId>
+    <artifactId>detekt-cli</artifactId>
+    <version>1.21.0</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:85]: ignore

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov @armiol Please take a look.

File: pull [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+cd ./config || exit
+echo "Switching the pointer of the 'config' submodule to its 'master'."
```
Comment (armiol) [inline pull:24]: This script performs different actions in different repos.

This version (in `change`) just updates the contents to the latest sources. But in other repos (as those with multiple copies of `buildSrc`) it also does other things as well.

We may include it into `template`, but in its simplest form (like the one here).

Also, in `1.x-dev` branches this script now looks differently, as it switches not to `master`, but to 1.x-specific branch of `config`. That's just for your information.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**